PlagiarismPlugin.php

```diff
@@ -86,206 +91,208 @@ function getEnabled($contextId = null)
-					$form->readInputData();
-					if ($form->validate()) {
-						$form->execute();
-						return new JSONMessage(true);
-					}
-				} else {
-					$form->initData();
-				}
-				return new JSONMessage(true, $form->fetch($request));
-		}
-		return parent::manage($args, $request);
-	}
-}
+        return array_merge(
+			$this->getEnabled() 
+				? [
+            		new LinkAction(
+						'settings',
+						new AjaxModal(
+							$router->url(
+								$request, 
+								null, 
+								null, 
+								'manage', 
+								null, 
+								[
+									'verb' 		=> 'settings', 
+									'plugin' 	=> $this->getName(), 
+									'category' 	=> 'generic'
+								]
+							),
+							$this->getDisplayName()
+						),
+						__('manager.plugins.settings'),
+						null
+            		),
+			 	] : [],
+                parent::getActions($request, $verb)
+        );
+    }
 
-/**
- * Low-budget mock class for \bsobbe\ithenticate\Ithenticate -- Replace the
- * constructor above with this class name to log API usage instead of
- * interacting with the iThenticate service.
- */
-class TestIthenticate
-{
-	public function __construct($username, $password)
+    /**
+     * @copydoc Plugin::manage()
+     */
+    function manage($args, $request) 
 	{
-		error_log("Constructing iThenticate: $username $password");
-	}
+        switch ($request->getUserVar('verb')) {
+            case 'settings':
+                $context = $request->getContext();
+                $templateMgr = TemplateManager::getManager($request);
+                $templateMgr->registerPlugin('function', 'plugin_url', [$this, 'smartyPluginUrl']);
+                $form = new PlagiarismSettingsForm($this, $context->getId());
 
-	public function fetchGroupList()
-	{
-		error_log('Fetching iThenticate group list');
-		return array();
-	}
+                if ($request->getUserVar('save')) {
+                    $form->readInputData();
+                    if ($form->validate()) {
+                        $form->execute();
+                        return new JSONMessage(true);
+                    }
+                } else {
+                    $form->initData();
+                }
 
-	public function createGroup($group_name)
-	{
-		error_log("Creating group named \"$group_name\"");
-		return 1;
-	}
-
-	public function createFolder($folder_name, $folder_description, $group_id, $exclude_quotes)
-	{
-		error_log("Creating folder:\n\t$folder_name\n\t$folder_description\n\t$group_id\n\t$exclude_quotes");
-		return true;
-	}
+                return new JSONMessage(true, $form->fetch($request));
+        }
 
-	public function submitDocument($essay_title, $author_firstname, $author_lastname, $filename, $document_content, $folder_number)
-	{
-		error_log("Submitting document:\n\t$essay_title\n\t$author_firstname\n\t$author_lastname\n\t$filename\n\t" . strlen($document_content) . " bytes of content\n\t$folder_number");
-		return true;
-	}
+        return parent::manage($args, $request);
+    }
 }
```

Comment:
It'll be necessary to add a `class_alias` for the old non-namespaced classname, since we still use the non-namespaced classname in `version.xml`. See e.g. the end of the `TinyMCEPlugin` class.

Reply:
added .